util/licence-checker.py

```diff
@@ -0,0 +1,357 @@
+    comment_style = detect_comment_char(filepath.name)
+
+    if comment_style is None:
+        results.skipped(filepath, "Unknown comment style")
+        return
+
+    if filepath.stat().st_size == 0:
+        results.skipped(filepath, "Empty file")
+        return
+
+    def next_line(file, line_no):
+        return (next(file).rstrip(), line_no + 1)
+
+    with filepath.open() as f:
+        licence_assumed_start = None
+
+        # Get first line
+        try:
+            line, line_no = next_line(f, 0)
+        except StopIteration:
+            results.failed(filepath, 1, "Empty file")
+            return
+
+        licence_search_marker = comment_style.search_line(licence.first_word)
+
+        # Check first line against the first word of licence, or against a
+        # possible different first line.
+        if not line.startswith(licence_search_marker):
+            if not line.startswith(comment_style.first_line_prefix):
+                results.failed(filepath, line_no,
+                               "File does not start with comment")
+                return
+
+            try:
+                line, line_no = next_line(f, line_no)
+            except StopIteration:
+                results.failed(filepath, line_no, "Reached end of file before finding licence")
+                return
+
+        # Skip lines that don't seem to be the first line of the licence
+        while not line.startswith(licence_search_marker):
+            try:
+                line, line_no = next_line(f, line_no)
+            except StopIteration:
+                results.failed(filepath, line_no, "Reached end of file before findling licence")
+                return
+
+            if not line.startswith(comment_style.comment_prefix):
+                results.failed(filepath, line_no,
+                               "First comment ended before licence notice")
+                return
+
+        # We found the marker, so we found the first line of the licence.
+        # The current line is in the first comment, so check the line matches the
+        # expected first line:
+        licence_assumed_start = line_no
+        if line != comment_style.expected_full_line(licence[0]):
+            results.failed(filepath, line_no, "Licence does not match")
+            return
+
+        for (licence_line_no, licence_line) in licence.numbered_lines(skip=1):
+
+            try:
+                line, line_no = next_line(f, line_no)
+            except StopIteration:
+                results.failed(filepath, line_no, "Reached end of file before finding licence")
+                return
+
+            # Check against full expected line.
+            if line != comment_style.expected_full_line(licence_line):
+                results.failed(filepath, line_no, "Licence did not match")
+                return
+
+    results.passed(filepath, licence_assumed_start, "Licence found")
+
+
+parser = argparse.ArgumentParser(
```

Comment:
I'd put everything below here in a `main` function, which should be called in the `if __name__ == '__main__'` block. That way, you don't put variables into the global namespace (which generally fools linting tools because of Python's weird scoping rules).